cmd/build.go

```diff
@@ -117,50 +123,90 @@ type flag struct {
 	Value         string
 }
 
+type generatorInput struct {
+	Version         uint
+	SourceDir       string
+	WorkingDir      string
+	BuildDirPrefix  string
+	BuildFlags      map[string]string
+	CompletionsOnly bool
+	RunArgs         []string
+}
+
 type generatorOutput struct {
+	Version   uint
 	NinjaFile string
-	BashFile  string
 	Targets   map[string]target
 	Flags     map[string]flag
 	BuildDir  string
 }
 
 var buildCmd = &cobra.Command{
-	Use:                   "build [targets] [build flags]",
-	Short:                 "Builds the targets",
-	Long:                  `Builds the targets.`,
-	Run:                   runBuild,
-	ValidArgsFunction:     completeBuildArgs,
+	Use:   "build [patterns] [build flags] [--commands] [--compdb] [--graph]",
+	Short: "Builds the targets",
+	Long:  `Builds the targets.`,
+	Run: func(cmd *cobra.Command, args []string) {
+		runBuild(args, false, []string{})
+	},
+	ValidArgsFunction: func(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+		return completeBuildArgs(toComplete, false), cobra.ShellCompDirectiveNoFileComp
+	},
 	DisableFlagsInUseLine: true,
 }
 
+var (
+	commandList     bool
+	commandDb       bool
+	dependencyGraph bool
+)
+
 func init() {
 	rootCmd.AddCommand(buildCmd)
+	buildCmd.Flags().BoolVar(&commandList, "commands", false, "Create compile commands list")
+	buildCmd.Flags().BoolVar(&commandDb, "compdb", false, "Create compile commands JSON database")
+	buildCmd.Flags().BoolVar(&dependencyGraph, "graph", false, "Create dependency graph")
 }
 
-func runBuild(cmd *cobra.Command, args []string) {
+func runBuild(args []string, runTargets bool, runArgs []string) {
 	dbtRulesDir := path.Join(util.GetWorkspaceRoot(), util.DepsDirName, dbtRulesDirName)
 	if !util.DirExists(dbtRulesDir) {
 		log.Fatal("You are running 'dbt build' without '%s' being available. Add that dependency, run 'dbt sync' and try again.\n", dbtRulesDirName)
 		return
 	}
 
-	targets, flags := parseArgs(args)
-	genOutput := runGenerator("buildFiles", flags)
+	patterns, flags := parseArgs(args)
+	genOutput := runGenerator(generatorInput{
+		BuildFlags: flags,
+		RunArgs:    runArgs,
+	})
 
-	// Write the build files.
+	// Determine the set of targets to be built.
+	log.Debug("Target patterns: '%s'.\n", strings.Join(patterns, "', '"))
+	regexps := []*regexp.Regexp{}
+	for _, pattern := range patterns {
+		re, err := regexp.Compile(fmt.Sprintf("^%s$", pattern))
+		if err != nil {
+			log.Fatal("Target pattern '%s' is not a valid regular expression: %s.\n", pattern, err)
+		}
+		regexps = append(regexps, re)
+	}
+	targets := []string{}
+	for target := range genOutput.Targets {
+		for _, re := range regexps {
+			if re.MatchString(target) {
+				targets = append(targets, target)
```

Comment:
Should also check that the target is runnable if they're trying to run it

Reply:
Good point. Done.